site_scons/site_tools/roc/config.py

```diff
@@ -290,6 +295,55 @@ def FindConfigGuess(context):
     context.Result('not found')
     return False
 
+def FindPkgConfigPath(context, crosscompile):
+    env = context.env
+
+    context.Message("Searching PKG_CONFIG_PATH...")
+
+    if env.HasArg('PKG_CONFIG_PATH'):
+        context.Result(env['PKG_CONFIG_PATH'])
+        return True
+
+    if env.HasArg('PKG_CONFIG'):
+        pkg_config = env['PKG_CONFIG']
+    elif not crosscompile and env.Which('pkg-config'):
+        pkg_config = 'pkg-config'
+    else:
+        pkg_config = ''
+
+    pkg_config_paths = env.CommandOutput('%s --variable pc_path pkg-config' % quote(pkg_config))
+    try:
+        for path in pkg_config_paths.split(':'):
+            if os.path.isdir(path):
+                env['PKG_CONFIG_PATH'] = path
+                break
+    except:
+        pass
+    finally:
+        if 'PKG_CONFIG_PATH' not in env.Dictionary():
+            # https://linux.die.net/man/1/pkg-config the default is libdir/pkgconfig
+            env['PKG_CONFIG_PATH'] = os.path.join(env['ROC_SYSTEM_LIBDIR'], 'pkgconfig')
+        context.Result(env['PKG_CONFIG_PATH'])
+    return True
+
+def AddPkgConfigDependency(context, crosscompile, package, flags):
+    env = context.env
+    if 'PKG_CONFIG_DEPS' not in env.Dictionary():
+        env['PKG_CONFIG_DEPS'] = []
+    if env.HasArg('PKG_CONFIG'):
+        pkg_config = env['PKG_CONFIG']
+    elif not crosscompile and env.Which('pkg-config'):
+        pkg_config = 'pkg-config'
```

Comment:
> I also added crosscompile check in AddPkgConfigDependency, let me know if it is desirable.

Good question!

After some reading, I think we should improve `pkg-config` tool detection algorithm. I suggest to extract it into another method, FindPkgConfig, and call it from SConstruct before calling FindPkgConfigPath and AddPkgConfigDependency.

I suggest the following algorithm:

* if `PKG_CONFIG` is set explicitly (check using HasArg), use it;
* if `toolchain` is not empty;

  * try to search for `toolchain + "-pkg-config"` in PATH (using Which); if it's found, use it;
  * otherwise, if `PKG_CONFIG_PATH`, or `PKG_CONFIG_LIBDIR`, or `PKG_CONFIG_SYSROOT_DIR` environment variable is set (check using HasArg), try to search for "pkg-config" in PATH; if it's found, use it (rationale: we will use system-wise pkg-config, but it will operate with cross-compilation sysroot, so that's OK);
  * otherwise, fail; `PKG_CONFIG` is not found;

* otherwise, if `toolchain` is empty, try to search for "pkg-config" in PATH; if it's found, use it;

* otherwise, fail; `PKG_CONFIG` is not found.

If `PKG_CONFIG` was not found, AddPkgConfigDependency should always return False. FindPkgConfigPath may still succeed, e.g. if PKG_CONFIG_PATH was set explicitly or if it fall back to default path.

Reference:

* https://autotools.io/pkgconfig/cross-compiling.html
* https://stackoverflow.com/questions/9221236/pkg-config-fails-to-find-package-under-sysroot-directory